omnigibson/simulator.py

```diff
@@ -1189,53 +1309,53 @@ def get_metadata(self, key):
             """
             return self.world_prim.GetCustomDataByKey(key)
 
-        def restore(self, json_path):
+        def restore(self, json_paths):
             """
-            Restore a simulation environment from @json_path.
+            Restore simulation environments from @json_paths.
 
             Args:
-                json_path (str): Full path of JSON file to load, which contains information
+                json_paths (List[str]): Full paths of JSON file to load, which contains information
                     to recreate a scene.
             """
-            if not json_path.endswith(".json"):
-                log.error(f"You have to define the full json_path to load from. Got: {json_path}")
-                return
+            for json_path in json_paths:
+                if not json_path.endswith(".json"):
+                    log.error(f"You have to define the full json_path to load from. Got: {json_path}")
+                    return
 
-            # Load the info from the json
-            with open(json_path, "r") as f:
-                scene_info = json.load(f)
-            init_info = scene_info["init_info"]
-            state = scene_info["state"]
+                # Load the info from the json
+                with open(json_path, "r") as f:
+                    scene_info = json.load(f)
+                init_info = scene_info["init_info"]
+                state = scene_info["state"]
 
-            # Override the init info with our json path
-            init_info["args"]["scene_file"] = json_path
+                # Override the init info with our json path
+                init_info["args"]["scene_file"] = json_path
 
-            # Also make sure we have any additional modifications necessary from the specific scene
-            og.REGISTERED_SCENES[init_info["class_name"]].modify_init_info_for_restoring(init_info=init_info)
+                # Also make sure we have any additional modifications necessary from the specific scene
+                og.REGISTERED_SCENES[init_info["class_name"]].modify_init_info_for_restoring(init_info=init_info)
 
-            # Recreate and import the saved scene
-            og.sim.stop()
-            recreated_scene = create_object_from_init_info(init_info)
-            self.import_scene(scene=recreated_scene)
+                # Recreate and import the saved scene
+                og.sim.stop()
+                recreated_scene = create_object_from_init_info(init_info)
+                self.import_scene(scene=recreated_scene)
 
-            # Start the simulation and restore the dynamic state of the scene and then pause again
-            self.play()
-            self.load_state(state, serialized=False)
+                # Start the simulation and restore the dynamic state of the scene and then pause again
+                self.play()
```

Comment:
maybe we can batch this to avoid N stop / play cycles?

Reply:
@cgokmen what do you think?

Reply:
Agreed, lets open a ticket for follow up and assign to me. I do have a partial implementation of this already.